### PR TITLE
Dynamic Validator Refactor for Operator Compatibility

### PR DESCRIPTION
**Pull request by bennerv**
Things to note: 
1. Removed the dependency from dynamic on any document specific types, so it should be able to be called from anywhere that matches the arguments for `NewValidator`.
2. OpenshiftCluster-validatedynamic still contains validation specific to the context of the RP.  If some of this functionality needs to be moved elsewhere, it would [ideally] move into dynamic.go.  
3. OpenshiftCluster-validatedynamic utilizes the caching mechanism of the virtualnetworkclient in some of it's vnet specific validation.  This can be changed or modified to use the raw `virtualNetworkClient` in `cache.go` instead of borrowing.  

### Which issue this PR addresses:
Refactor openshiftcluster-dynamicvalidator into an interface that allows some functions to be used outside of the context of the RP.  

### What this PR does / why we need it:
This should allow us to call validation methods from the context of the operator to expose them as conditions.  

### Test plan for issue:
Unit tests 
E2E tests

### Is there any documentation that needs to be updated for this PR?
No

### CODE REVIEW

**Comment by mjudeikis** [comment]
I think this makes code more re-usable but it makes all the code exponentially more complicated and harder to read. Im ok with this but we just need to be more aware of things like this. Now it requires more experienced reviews, code is harder to change, any task touching this code will be much harder. Just saying :)

File: pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml [hunk 1]
```diff
@@ -69,7 +69,7 @@ spec:
             resourceId:
               description: ResourceID is the Azure resourceId of the cluster
               type: string
-            vnetId:
+            vnetID:
```
Comment (mjudeikis) [inline pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml:72]: vnetId

File: pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go [hunk 2]
```diff
@@ -39,8 +39,9 @@ type ClusterSpec struct {
 	Location        string              `json:"location,omitempty"`
 	GenevaLogging   GenevaLoggingSpec   `json:"genevaLogging,omitempty"`
 	InternetChecker InternetCheckerSpec `json:"internetChecker,omitempty"`
-	VNetID          string              `json:"vnetId,omitempty"`
-	Features        FeaturesSpec        `json:"features,omitempty"`
+	VnetID          string              `json:"vnetID,omitempty"`
```
Comment (mjudeikis) [inline pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go:42]: json:"vnetId,omitempty"`

File: pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml (hunk 1 above)
Comment (bennerv) [reply pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml]: Fixed

File: pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go (hunk 2 above)
Comment (bennerv) [reply pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go]: Fixed

**Review by mjudeikis — APPROVED** [review]

File: pkg/api/validate/dynamic.go [hunk 3]
```diff
@@ -43,13 +45,15 @@ type dynamic struct {
 	oc    *api.OpenShiftCluster
 	vnetr *azure.Resource
 
+	code string
+	typ  string
```
Comment (mjudeikis) [inline pkg/api/validate/dynamic.go:46]: Please raise follow-up to try to get this generalized so this would be set by called instead in the initiation

**Pull request merged**